### PR TITLE
add test case: delete resources controlled by tc

### PR DESCRIPTION
**Pull request by dragonly**
Signed-off-by: Li Yilong <liyilongko@gmail.com>

### What problem does this PR solve?

Add e2e case: delete resources controlled by tc.

### Code changes

- [ ] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [ ] Unit test 
- [x] E2E test 
- [ ] Manual test 
- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [x] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note
None
```

### CODE REVIEW

File: tests/e2e/tidbcluster/tidbcluster.go [hunk 1]
```diff
@@ -1558,6 +1559,38 @@ var _ = ginkgo.Describe("TiDBCluster", func() {
...
 		oa.CheckTidbClusterStatusOrDie(&tcCfg)
 	})
+
+	ginkgo.It("Deleted objects controlled by TidbCluster will be recovered by Operator", func() {
+		ginkgo.By("Deploy initial tc")
+		tc := fixture.GetTidbCluster(ns, "delete-objects", utilimage.TiDBV4)
+		utiltc.MustCreateTCWithComponentsReady(genericCli, oa, tc, 5*time.Minute, 10*time.Second)
+
+		ginkgo.By("Delete StatefulSet/ConfigMap/Service of PD")
+		deleteResourcesForComponent := func(memberName string) {
```
Comment (DanielZhangQD) [inline tests/e2e/tidbcluster/tidbcluster.go:1569]: What about using label selector `app.kubernetes.io/instance` to delete the resources without caring about the components?

**Review by DanielZhangQD — APPROVED** [review]
LGTM

**Review by handlerww — APPROVED** [review]
LGTM

**Pull request merged**